data-prepper-plugins/aws-lambda/src/main/java/org/opensearch/dataprepper/plugins/lambda/processor/LambdaProcessor.java

```diff
@@ -364,6 +374,73 @@ private void checkCircuitBreaker() {
         }
     }
 
+    private Collection<Record<Event>> processWithStreaming(
+            List<Record<Event>> recordsToLambda, 
+            List<Record<Event>> resultRecords) {
+        
+        try {
+            checkCircuitBreaker();
+        } catch (Exception e) {
+            LOG.error(NOISY, "Circuit breaker check failed", e);
+            numberOfRecordsFailedCounter.increment(recordsToLambda.size());
+            numberOfRequestsFailedCounter.increment();
+            resultRecords.addAll(addFailureTags(recordsToLambda));
+            return resultRecords;
+        }
+
+        InputCodec responseCodec = pluginFactory.loadPlugin(InputCodec.class, codecPluginSetting);
```

Comment:
Generating a new plugin for every invocation will be expensive.

Also, why can't we use the existing response codec that the non-streaming implementation uses? I think there is no need for a different instance.